### ISSUE TITLE
transitions with appear should be rendered with the "-enter" class

### ISSUE THREAD

**Issue by matthieusieben** [vuejs/vue#10590]
### Version
2.6.10

### Reproduction link
[https://jsfiddle.net/chrisvfritz/50wL7mdz/](https://jsfiddle.net/chrisvfritz/50wL7mdz/)

### Steps to reproduce
Create a ssr page containing `<transition appear>` elements. Load the page and notice a flickering of the animation.

### What is expected?
The server response should contain the `*enter` classes applied to the transitioning element. The `*active` classes should then be applied during the `hydration` so that the transition can appear smoothly.

### What is actually happening?
The element is first displayed, as if there were no transitions. Then then entire animation starts, causing wired animations.

**Comment by matthieusieben** [comment]
This issue is still unresolved.

**Comment by milesingrams** [comment]
I have this issue too. Expecting appear transitions to have the appear-from class applied on static generation and SSR so the initial position can be appropriately rendered until vue mounts

**Comment by drewbaker** [comment]
Yeah I expected that `appear` works for initial animation when SSR or client side render occurs. Otherwise not sure the point of appear, if I need to toggle a v-if on mounted to get the initial animation to happen anyway.
